spec/models/vote_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Vote, type: :model do
+  describe "Associations for vote on post" do
+    subject { build(:up_vote) }
+
+    it { should belong_to(:user) }
+    # it { should belong_to(:post) }
+    # error says no foreign key
+  end
+
+  describe "Associations for vote on comment" do
+    subject { build(:up_vote_comment) }
+
+    it { should belong_to(:user) }
+    # it { should belong_to(:comment) }
```

Comment:
This should be covered by the above. This part definitely isn't necessary, but if you also want to test that both comments an votes can be votables, that's outside the scope of shouldamatchers and you'll need to do it manually. Something like:
```ruby
let(:post) { build(:post) } 
let(:comment) { build(:comment) } 
let(:post_vote) { build(:vote, votable: post }
let(:comment_vote) { build(:vote, votable: comment }

it "can have a post or comment as a votable" do
  expect(post_vote.votable).to eq(post)
  expect(comment_vote.votable).to eq(comment)
end
```
or, using your factories a little more:
```ruby
let(:post_vote) { build(:up_vote) }
let(:comment_vote) { build(:up_vote_comment) }

it "can have a post or comment as a votable" do
  expect(post_vote.votable).to be_a(Post)
  expect(comment_vote.votable).to be_a(Comment)
end
```
I didn't actually run any of that ^ so it mightbe a little off.